statsd/aggregator.go

```diff
@@ -223,6 +230,12 @@ func (a *aggregator) gauge(name string, value float64, tags []string) error {
 	gauge := newGaugeMetric(name, value, tags)
```

Comment:
Silly question since I didn't dug much into datadog-go but, are you sure that you are winning anything from this `RUnlock` if you are re-locking the `gaugesM` right after that `newGaugeMetric` call? That question is true for all your changes since it's the same everywhere.

Reply:
We need to `RUnlock` to be able to `Lock` right after.

Reply:
My question is more about: is it worth unlocking then? Keeping the lock while running `newGaugeMetric` may not negatively impact performances since you actually try to get that lock right after again.

Reply:
Discussed offline: the first RLock lets other routine sample the same metrics, so it can't be switched to Lock for performances reasons.